### ISSUE TITLE
refine() uses softmax outputs as parent logits, breaking HVAQ temperature scaling

### ISSUE THREAD

**Issue by sachncs** [sachncs/avqa#14]
## Problem

In `src/avqa/refinement.py:233`, the refinement step gathers what the function calls "parent logits" from `parent_probs`:

```python
parent_logit_gathered = parent_probs.gather(-1, selected.unsqueeze(-2).expand(B, H, T, P))
```

But `parent_probs` is the **softmax-normalized** attention distribution returned by `online_softmax` (see `src/avqa/pipeline.py:332-334`), **not** the raw dot-product logits. `parent_probs[i, j] ∈ [0, 1]` and sums to 1 along the M₀ axis. The subsequent `vectorized_correction` then exponentiates these "logits" and re-softmaxes them in `refinement.py:139-147`, which silently rescales the correction.

## Evidence

`src/avqa/refinement.py:233`:

```python
parent_logit_gathered = parent_probs.gather(-1, selected.unsqueeze(-2).expand(B, H, T, P))
```

`src/avqa/pipeline.py:332-334`:

```python
softmax_state, parent_attention_probs = online_softmax(
    parent_logits_t, result.parent_aggregates, result.parent_counts, D, D_v,
)
```

The variable name `parent_attention_probs` documents that this tensor is **probabilities**, not logits. The `refine` docstring (`refinement.py:189-190`) also documents `parent_probs: [B, H, T, M_0] parent attention probabilities` — but the implementation then treats it as logits inside `vectorized_correction`.

The downstream exponentiation in `vectorized_correction` (`refinement.py:139-147`) does:

```python
parent_exp = torch.exp(parent_logit - m_anchor) * parent_scale
parent_contrib_denom = parent_exp.sum(dim=-1, keepdim=True)
```

…where `parent_logit` is now `(softmax output) ∈ [0,1]`. After `exp`, this yields values in `[1, e]` rather than the original logits' `[e^(-inf), e^(max_logit)]` range. The correction's softmax denominator is therefore distorted by a factor of `e^(-max_logit)`, which the `m_anchor` rescale can only partially cancel.

## Impact

- With HVAQ off and `config.tolerance_atol=1e-5`, the discrepancy is small (softmax inputs are in a narrow range), so existing tests pass.
- With HVAQ on (multiplied by a per-query β_q), the discrepancy compounds: the HVAQ scale enters the logits **before** softmax, but the refinement then sees softmax(output), not output. The HVAQ schedule therefore does not produce the documented `β_q · β_p · (q · p^T)` formula — the `β_q · β_p` is applied to the post-softmax probabilities.
- BCAR / online adaptation can also be affected since the correction state diff is propagated through the (already-scaled) probability channel.
- Users who rely on the spec-claimed invariant (HVAQ temperature monotonicity, Theorem 16.2) will see non-monotone behavior in some input regimes.

## Root Cause

The function signature was documented as taking `parent_probs` (the post-softmax output) but the implementation requires **pre-softmax parent logits**. The mismatch was masked because the typical test inputs have low magnitude (random `N(0, 1)`), keeping both in similar ranges.

## Recommendation

Pass the **raw logits** through the refinement call. Two viable approaches:

1. Capture the pre-softmax parent logits in `online_softmax` (return them alongside `parent_attention_probs`).
2. Re-compute parent logits in `refine` from `q` and `state.codebook.parents` (`logits = q @ parents^T / sqrt(D)`) at the cost of an extra matmul.

Option 1 is the cheaper fix and preserves determinism. Both will require updating the public `refine` signature and the test fixtures that pass synthetic `parent_probs`.

## Acceptance Criteria

- [ ] `refine(...)` no longer treats softmax probabilities as raw logits.
- [ ] Add a property test (Hypothesis) that asserts `refine` is invariant under positive rescaling of the parent logits: `refine(parent_logits + c) == refine(parent_logits)` for scalar `c`.
- [ ] HVAQ Theorem 16.2 ("top-K index invariance under positive β") test in `tests/unit/test_hopfield.py` is extended to cover the refinement path.
- [ ] Existing `tests/unit/test_refinement.py` continue to pass.

## Verification

Run the HVAQ downstream-consumer invariant test on a non-uniform distribution and confirm the top-1 parent index is invariant under `β_q ∈ [0.1, 10]` (currently fails when the refinement path uses softmax outputs).

## Risk

Medium. The behavior is observable in HVAQ mode. Fixing this will change the documented correction formula; downstream users that worked around the bug need to re-verify their results.

**Comment by sachncs** [comment]
Resolved in PR #37.
